src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -291,7 +291,7 @@ private void fillNextSelectedKeysInPartition(DecoratedKey partitionKey, List<Pri
                 if (key == null)
                     break;
 
-                if (queryController.doesNotSelect(key) || key.equals(lastKey))
+                if (queryController.doesNotSelect(key) || key.equals(lastKey, false))
```

Comment:
The idea with these two changes is that, if we've already processed a STATIC key in a partition, there is no need to process any normal rows in the same partition (because we'll have post-filtered the whole partition already).